### PR TITLE
[fluentd] Pin gem versions

### PR DESCRIPTION
**Pull request by periklis**
### Motivation
Pins current dependencies as mentioned in `rh-manifest.txt`.

### CODE REVIEW

File: hack/update-fluentd-vendor-gems.sh [hunk 1]
```diff
@@ -68,17 +68,10 @@ while read gemname gemver ; do
...
-        fi
-        if foundname=$( brew search package --exact $pkgname ) && [ "$foundname" = "$pkgname" ] ; then
-            : # package exists in brew
-        else
-            pkgname=$gemname
-        fi
+    if [ $gemname = fluentd ] ; then
+        pkgname=fluentd
+    else
+        pkgname="rubygem-${gemname}"
```
Comment (jcantrill) [inline hack/update-fluentd-vendor-gems.sh]: do we need this?  This I think is a holdover of gems being packaged in RPMs which would be unnecessary if we are vendoring and no longer relying on brew

Comment (periklis) [reply hack/update-fluentd-vendor-gems.sh]: OMG, i thought this is a hindsight of someone using brew on a macOS machine.

Comment (periklis) [reply hack/update-fluentd-vendor-gems.sh]: I fixed this script for local dev machines, where brew is missing.

File: fluentd/Gemfile [hunk 2]
```diff
@@ -1,14 +1,14 @@
 gem 'fluentd', ENV['FLUENTD_VERSION']
```
Comment (jcantrill) [inline fluentd/Gemfile]: please remove the env var here and from the Dockerfiles and we'll peg the version in the Gemfile too

Comment (periklis) [reply fluentd/Gemfile]: AFAICS, `ENV['FLUENTD_VERSION']` is used heavily in the source code. I hesitate to delete this. We should keep the version pin in the dockerfile. WDYT?

**Pull request merged**